Public/Src/Tools/Xldb.Analyzer/XldbAnalyzer.cs

```diff
@@ -0,0 +1,314 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using Xldb.Proto;
+
+namespace Xldb.Analyzer
+{
+    /// <summary>
+    /// Xldb Analyzers that use the rocksDb instanceand the Xldb API to analyze BXL logs
+    /// </summary>
+    public class Program
+    {
+
+        private const string s_eventStatsAnalyzer = "eventstats";
+        private const string s_dumpPipAnalyzer = "dumppip";
+
+        private Dictionary<string, string> m_commandLineOptions = new Dictionary<string, string>();
+
+        public static int Main(string[] args)
+        {
+            var p = new Program();
+
+            if (!p.ProcessArguments(args))
+            {
+                Console.WriteLine("Invalid arguments passed in, exiting analyzer ...");
+                return 1;
+            }
+
+            if (p.m_commandLineOptions.TryGetValue("/m", out var mode))
+            {
+                if (mode == s_eventStatsAnalyzer)

```

Comment:
```suggestion
                if (string.Equals(s_eventStatsAnalyzer, mode, StringComparer.OrdinalIgnoreCase))
```

to handle different casing on the commandline

Reply:
alternatively if you prefer you could write a switch statement because C# supports a switch on strings like:

```
switch (mode.ToLowerInvariant())
{
    case s_eventSTatsAnalyzer:
        return p.AnalyzeEventStats();
    case s_dumpPipAnalyzer:
        return p.AnalyzeDumpPip();
     default:
           Console.Write($"Valid modes are: /m:{s_eventStatsAnalyzer}, /m:{s_dumpPipAnalyzer}");
           return 1;
}
```

---
In reply to: [315939965](https://github.com/microsoft/BuildXL/pull/775#discussion_r315939965) [](ancestors = 315939965)

Reply:
Sure, i can use a switch statement. I left the "valid modes" print after the if statement though so that I would not have to repeat as much code and let both the errors (bad mode, or missing mode) get caught with 1 print statement.

---
In reply to: [315940523](https://github.com/microsoft/BuildXL/pull/775#discussion_r315940523) [](ancestors = 315940523,315939965)